File: generalgraph/naivebfs.cpp
#include <iostream>
#include <fstream>
#include <functional>
#include <vector>
#include <queue>
#include <utility>
using namespace std;

int main() {
	ifstream in("input.in");
	ofstream out("output.out");

	auto my_assert = [&](bool cond, const char* comment) {
		if (!cond) {
			out << "Assertion Failed: " << comment << '\n';
			out.close();
			exit(0);
		}
	};
	
	int graph_size, edge_num;
	in >> graph_size >> edge_num;
	vector<vector<int>> adj(graph_size);
	for (int i = 0; i < edge_num; i++) {
		int u, v;
		in >> u >> v;
		--u, --v;
		my_assert(u != v && 0 <= u && u < graph_size && 0 <= v && v < graph_size, "wrong input");
		adj[u].push_back(v);
		adj[v].push_back(u);
	}
	vector<int> ecc(graph_size);

	for (int i = 0; i < graph_size; i++) {
		queue<pair<int, int>> qu;
		vector<bool> visited(graph_size);
		qu.emplace(i, 0);
		visited[i] = true;
		while (!qu.empty()) {
			auto [from, dist] = qu.front(); qu.pop();
			ecc[i] = max(ecc[i], dist);
			for (int to : adj[from]) {
				if (!visited[to]) {
					qu.emplace(to, dist + 1);
					visited[to] = true;
				}
			}
		}
	}

	vector<int> cnt(graph_size);
	for (int i = 0; i < graph_size; i++) {
		my_assert(0 <= ecc[i] && ecc[i] < graph_size, "invalid eccentricity");
		cnt[ecc[i]]++;
	}
	vector<int> eccseq;
	for (int i = 0; i < graph_size; i++) {
		for (int j = 0; j < cnt[i]; j++) {
			eccseq.push_back(i);
		}
	}
	for (int i = 0; i < eccseq.size(); i++) {
		out << eccseq[i] << ' ';
	}
	out << '\n';
	out.close();
	return 0;
}

File: generalgraph/naivebfs.py
import sys, os
from collections import deque

sys.setrecursionlimit(10 ** 6)
__location__ = os.path.realpath(os.path.join(os.getcwd(), os.path.dirname(__file__)))
inf = open(os.path.join(__location__, "input.in"), "r")
ouf = open(os.path.join(__location__, "output.out"), "w")

def my_assert(cond, comment):
    if not cond:
        ouf.write(f"Assertion Failed: {comment}")
        ouf.close()
        sys.exit(0)

graph_size, edge_num = map(int, inf.readline().rstrip().split())
adj = [[] for _ in range(graph_size)]
for i in range(edge_num):
    u, v = map(int, inf.readline().rstrip().split())
    u -= 1; v -= 1
    my_assert(u != v and 0 <= u < graph_size and 0 <= v < graph_size, "wrong input")
    adj[u].append(v)
    adj[v].append(u)
inf.close()
ecc = [0] * graph_size

for i in range(graph_size):
    qu = deque()
    visited = [False] * graph_size
    qu.append((i, 0))
    visited[i] = True
    while qu:
        cur, dist = qu[0]
        qu.popleft()
        ecc[cur] = max(ecc[cur], dist)
        for next in adj[cur]:
            if not visited[next]:
                qu.append((next, dist + 1))
                visited[next] = True

cnt = [0] * graph_size
for i in range(graph_size):
    my_assert(0 <= ecc[i] < graph_size, "invalid eccentricity")
    cnt[ecc[i]] += 1
eccseq = []
for i in range(graph_size):
    for j in range(cnt[i]):
        eccseq.append(i)
for i in range(len(eccseq)):
    ouf.write(f"{eccseq[i]} ")
ouf.write("\n")
ouf.close()